src/services/api/routes/communities.ts

```diff
@@ -28,22 +28,48 @@ const createNewCommunitySchema = z.object({
 
 const router: express.Router = express.Router();
 
+async function respondCommunityNotFound(response: express.Response) : Promise<void> {
+	response.status(404);
+	response.send(xmlbuilder.create({
+		result: {
+			has_error: 1,
+			version: 1,
+			code: 404,
+			error_code: 919,
+			message: 'COMMUNITY_NOT_FOUND'
+		}
+	}).end({ pretty: true }));
+}
+
+async function commonGetSubCommunity(request: express.Request, response: express.Response, communityID: string | undefined) : Promise<HydratedCommunityDocument | null> {
+
+	const parentCommunity: HydratedCommunityDocument | null = await getCommunityByTitleID(request.paramPack.title_id);
+	if (!parentCommunity) {
+		await respondCommunityNotFound(response);
+		return parentCommunity;
+	}
+
+	const query: SubCommunityQuery = {
+		parent: parentCommunity.olive_community_id,
+		community_id: communityID
+	};
+
+	const community: HydratedCommunityDocument | null = await Community.findOne(query);
+	if (!community) {
+		await respondCommunityNotFound(response);
+		return community;
+	}
+
+	return community;
+}
```

Comment:
This function is only used once, is there a reason this was not just done in the route handler instead?